File: src/main/java/com/dh/ondot/member/domain/OauthInfo.java
package com.dh.ondot.member.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Embeddable
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class OauthInfo {
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private OauthProvider oauthProvider;

    @Column(nullable = false)
    private String oauthProviderId;

    public static OauthInfo of(OauthProvider oauthProvider, String oauthProviderId) {
        return new OauthInfo(oauthProvider, oauthProviderId);
    }
}

File: src/main/kotlin/com/dh/ondot/member/domain/OauthInfo.kt
package com.dh.ondot.member.domain

import com.dh.ondot.member.domain.enums.OauthProvider
import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

@Embeddable
class OauthInfo(
    @Enumerated(EnumType.STRING)
    @Column(name = "oauth_provider", nullable = false)
    val oauthProvider: OauthProvider,

    @Column(name = "oauth_provider_id", nullable = false)
    val oauthProviderId: String,
) {
    companion object {
        fun of(oauthProvider: OauthProvider, oauthProviderId: String): OauthInfo =
            OauthInfo(oauthProvider, oauthProviderId)
    }
}
